staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -270,15 +274,41 @@ func (r *ScaleREST) Update(ctx context.Context, name string, objInfo rest.Update
 	}
 
 	specReplicasPath := strings.TrimPrefix(r.specReplicasPath, ".") // ignore leading period
-	if err = unstructured.SetNestedField(cr.Object, int64(scale.Spec.Replicas), strings.Split(specReplicasPath, ".")...); err != nil {
-		return nil, false, err
+	var crObjInfo rest.UpdatedObjectInfo
+	if len(scale.ResourceVersion) == 0 {
+		// No resourceVersion was provided in the request.
+		// This means the client expects the scale update to take effect
+		// even if a concurrent update to the custom resource was made.
+
+		// We may not want to set the resourceVersion manually because
+		// it may be changed by other concurrent updates,
+		// nor retry on conflicts because it is possible that we keep
+		// retrying until timeout in extreme cases like frequent concurrent patches.
+		// As a result, we do a service-side patch here.
+		crObjInfo = rest.DefaultUpdatedObjectInfo(nil, func(_ context.Context, _, currentObj runtime.Object) (runtime.Object, error) {
```

Comment:
actually, I think you're on the right track with this change, but it doesn't go far enough.

we should move almost all of the function into a DefaultUpdatedObjectInfo impl, and stop doing our own `r.store.Get` outside the Update call

Reply:
@liggitt done! PTAL